src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1686,9 +1687,9 @@ abstract class ManagedStrategy(
         }
 
         if (isInTraceDebuggerMode && isFirstReplay && deterministicMethodDescriptor != null) {
-            deterministicMethodDescriptor.saveFirstResultWithCast(receiver, params, KResult.success(result)) {
+            newResult = deterministicMethodDescriptor.saveFirstResultWithCast(receiver, params, KResult.success(result)) {
                 nativeMethodCallStatesTracker.setState(descriptorId, deterministicMethodDescriptor.methodCallInfo, it)
-            }
+            }.getOrThrow()
```

Comment:
Isn't it too early to exit?
What about other stuff that happens below?

Reply:
I expect it to never happen. Errors should not be replaced with not errors, etc.